applications/ConstitutiveLawsApplication/custom_constitutive/small_strains/plastic_damage/associative_plastic_damage_model.cpp

```diff
@@ -470,6 +551,71 @@ void AssociativePlasticDamageModel<TYieldSurfaceType>::CalculateThresholdAndSlop
                 rPDParameters.Slope     = CalculateSlopeFiniteDifferences(implicit_function, function_derivative, rValues, rPDParameters, max_threshold*limit_factor);
                 break;
             }
+            case GenericConstitutiveLawIntegratorPlasticity<TYieldSurfaceType>::HardeningCurveType::CurveDefinedByPoints: {
+                const double xi = rPDParameters.PlasticDamageProportion;
+                const double g = CalculateVolumetricFractureEnergy(rValues.GetMaterialProperties(), rPDParameters);
+                const double C0 = rValues.GetMaterialProperties()[YOUNG_MODULUS];
+                double K0;
+                GenericConstitutiveLawIntegratorPlasticity<TYieldSurfaceType>::GetInitialUniaxialThreshold(rValues, K0);
+                const double E0 = K0 / C0; // Yield strain
+                const Vector& s_by_points = rValues.GetMaterialProperties()[EQUIVALENT_STRESS_VECTOR_PLASTICITY_POINT_CURVE]; // By-Points region. Stress vector
+                const Vector& e_by_points = rValues.GetMaterialProperties()[TOTAL_STRAIN_VECTOR_PLASTICITY_POINT_CURVE]; // By-Points region. Strain vector
```

Comment:
to avoid repetition, you can create a `const auto& r_props = rValues.GetMaterialProperties();` and reuse it